code/backend/batch/utilities/helpers/DocumentProcessorHelper.py

```diff
@@ -22,20 +28,38 @@ def __init__(self):
         pass
 
     def process(self, source_url: str, processors: List[Processor]):
-        vector_store_helper = AzureSearchHelper()
-        vector_store = vector_store_helper.get_vector_store()
-        for processor in processors:
+        env_helper: EnvHelper = EnvHelper()
+        if env_helper.AZURE_SEARCH_USE_INTEGRATED_VECTORIZATION:
             try:
-                document_loading = DocumentLoading()
```

Comment:
Am I right in saying that this code runs every time a new file is uploaded? Yet, the code doesn't use `source_url` at all

It seems to me like we would only want this code to run one time, once the infra has been provisioned, not every time a file is uploaded?

Reply:
Or is it that we manually kick off the indexer every time a file is uploaded?

Reply:
This is the existing method for Push approach. Is called everytime a document is uploaded